bundles/binding/org.openhab.binding.plclogo/src/main/java/org/openhab/binding/plclogo/internal/PLCLogoBinding.java

```diff
@@ -0,0 +1,562 @@
+
+            lock.lock();
+            int result = LogoS7Client.ReadArea(S7.S7AreaDB, 1, address, size, buffer);
+            logger.debug("Read word from logo memory: at {} {} bytes, result = {}", address, size, result);
+            if (result == 0) {
+                Item item = config.getItem();
+                if (item instanceof NumberItem && !wr.isDigital()) {
+                    if (command instanceof DecimalType) {
+                        int oldValue = S7.GetShortAt(buffer, 0);
+                        int newValue = ((DecimalType) command).intValue();
+                        S7.SetWordAt(buffer, 0, newValue);
+                        logger.debug("Changed word at {} from {} to {}", address, oldValue, newValue);
+
+                        result = LogoS7Client.WriteArea(S7.S7AreaDB, 1, address, size, buffer);
+                        logger.debug("Wrote to memory at {} two bytes: [{}, {}]", address, buffer[0], buffer[1]);
+                    }
+                } else if (item instanceof SwitchItem && wr.isDigital()) {
+                    if (command instanceof OnOffType) {
+                        boolean oldValue = S7.GetBitAt(buffer, 0, bit) ? true : false;
+                        boolean newValue = command == OnOffType.ON ? true : false;
+                        S7.SetBitAt(buffer, 0, bit, newValue);
+                        logger.debug("Changed bit {}.{} from {} to {}", address, bit, oldValue, newValue);
+
+                        result = LogoS7Client.WriteArea(S7.S7AreaDB, 1, address, size, buffer);
+                        logger.debug("Wrote to memory at {} one byte: [{}]", address, buffer[0]);
+                    }
+                }
+
+                // If nothing was written and read was ok, nothing will happen here
+                if (result != 0) {
+                    logger.warn("Failed to write memory: {}. Reconnecting...", S7Client.ErrorText(result));
+                    ReconnectLogo(LogoS7Client);
+                }
+            } else {
+                logger.warn("Failed to read memory: {}. Reconnecting...", S7Client.ErrorText(result));
+                ReconnectLogo(LogoS7Client);
+            }
+            lock.unlock();
+        }
+    }
+
+    /**
+     * @{inheritDoc}
+     */
+    @Override
+    protected void internalReceiveUpdate(String itemName, State newState) {
+        // the code being executed when a state was sent on the openHAB
+        // event bus goes here. This method is only called if one of the
+        // BindingProviders provide a binding for the given 'itemName'.
+        super.internalReceiveUpdate(itemName, newState);
+        logger.debug("internalReceiveUpdate() is called!");
+    }
+
+    /**
+     * @{inheritDoc}
+     */
+    @Override
+    public void updated(Dictionary<String, ?> config) throws ConfigurationException {
+        Boolean configured = false;
+        if (config != null) {
+            String refreshIntervalString = (String) config.get("refresh");
```

Comment:
If you plan on adding configuration via the Paper UI in openHAB 2, replace String casting of the config dictionary with `Objects.toString(config.get("X"), null);`, since the config might have other types besides String, like Integer and Boolean.